### DISCUSSION TITLE
Use json/yaml/toml to replace meson.options?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [mesonbuild/meson#13138] [Ideas]
Is it possible to use a data exchange format to replace meson.options? They are also not Turing Complete and easy to learn like meson lang.

No offense. Just some thought. :smile:

**Comment by eli-schwartz** [comment]
Unlikely IMO.

The current options file is carefully designed to not support the flexibility of the full meson DSL, so I'm not sure what you mean regarding turing completeness.

That being said, meson.build isn't turing complete either. It has control flow, math, assignment, object-oriented design, etc. but it is decidable and guaranteed to run within the constraints of finite time, which is already a disqualification. More to the point, it doesn't have functions, and cannot implement recursion.

**Comment by eli-schwartz** [comment]
Json isn't a great config format to use because it lacks comment support.

Yaml isn't an option because it can't be done via the python standard library.

Toml could be done with the python standard library, but only by dropping support for versions of python less than 3.11; our current minimum requirement is 3.7 and we aren't going to drop support for non-EOL pythons anyway. But also I hate toml :p

If we were going to choose something it would be INI, which meson already uses in several places such as the wrap file format and machine files.

**Reply by Freed-Wu** [reply]
Oh, I forget INI. Perhaps meson.options can use INI?
